storage-node/packages/runtime-api/workers.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of the storage node for the Joystream project.
+ * Copyright (C) 2019 Joystream Contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+'use strict'
+
+const debug = require('debug')('joystream:runtime:roles')
+const BN = require('bn.js')
+// const { createType } = require('@polkadot/types')
+const { Worker } = require('@joystream/types/lib/working-group')
+
+/*
+ * Add worker related functionality to the substrate API.
+ */
+class WorkersApi {
+  static async create (base) {
+    const ret = new WorkersApi()
+    ret.base = base
+    await ret.init()
+    return ret
+  }
+
+
+  // eslint-disable-next-line class-methods-use-this, require-await
+  async init () {
+    debug('Init')
+  }
+
+  /*
+   * Check whether the given account and id represent an active storage provider
+   */
+  async isRoleAccountOfStorageProvider (storageProviderId, roleAccountId) {
+    const id = new BN(storageProviderId)
+    const roleAccount = this.base.identities.keyring.decodeAddress(roleAccountId)
+    const providerAccount = await this.storageProviderRoleAccount(id)
+    return providerAccount && providerAccount.eq(roleAccount)
+  }
+
+  async isStorageProvider (storageProviderId) {
+    const worker = await this.storageWorkerByProviderId(storageProviderId)
+    return worker !== null
+  }
+
+  // Returns a provider's role account or null if provider doesn't exist
+  async storageProviderRoleAccount (storageProviderId) {
+    const worker = await this.storageWorkerByProviderId(storageProviderId)
+    return worker ? worker.role_account : null
+  }
+
+  // Returns a Worker instance or null if provider does not exist
+  async storageWorkerByProviderId (storageProviderId) {
+    const id = new BN(storageProviderId)
+    const { providers } = await this.getAllProviders()
+    return providers[id.toNumber()] || null
+  }
+
+  async findProviderIdByRoleAccount (roleAccount) {
+    const { ids, providers } = await this.getAllProviders()
+
+    for (let i = 0; i < ids.length; i++) {
+      const id = ids[i]
+      if (providers[id].role_account.eq(roleAccount)) {
+        return id
+      }
+    }
+
+    return null
+  }
+
```

Comment:
Thats an interesting strategy. We have casually considered in the past to add custom rpc calls to query state, especially complex queries. 

> Do we include the '.eslintrc.js' rules alteration in the scope of the PR?

I was originally thinking of making both formatting and esliting fixes in a separate PR, so would include modifying the rules also. Maybe worth doing it in this PR now that you mention it.

Reply:
And I can see your point of preserving older nodes in the working state. Probably we could release such 'custom rpc optimizations' in a bunch during the node upgrade, but it would also make the system more fragile because we would need to do more frequent node upgrades than necessary. Ok, then we leave it until we really need it in order to avoid some time & resources consuming iteration through the storage node.